crates/notion-core/src/package_info.rs

```diff
@@ -0,0 +1,102 @@
+//! Provides the `PackageInfo` type, which contains data for a Node project (from `package.json`).
+
+use std::collections::HashMap;
+use std::fs::File;
+use std::path::Path;
+
+use notion_fail::{Fallible, ResultExt};
+use serde_json;
+
+use serial;
+
+/// Info about a Node package
+pub struct PackageInfo {
```

Comment:
I made this separate because serializing `Manifest` requires the `"notion"` section to be present in package.json. Which doesn't work for dependencies, because they won't have that section. So instead of trying to rework how that is handled, and change everywhere it's used, I opted to create a new one for this purpose.

Reply:
I'm a little surprised that would be hard to reconcile, but if this was the easiest way for you to get things working for your immediate needs, I'm fine with it for now. As followup, I think we should be able to merge the two types and make the `"notion"` info optional.